src/support/scratch.c

```diff
@@ -155,6 +155,89 @@ __wt_buf_set_printable(
 	return (buf->data);
 }
 
+/*
+ * __wt_buf_set_printable_format --
+ *	Set the contents of the buffer to a printable representation of a byte
+ * string, based on a format.
+ */
+const char *
+__wt_buf_set_printable_format(WT_SESSION_IMPL *session,
+    const void *p, size_t size, const char *format, WT_ITEM *buf)
+{
+	WT_DECL_ITEM(tmp);
+	WT_DECL_PACK_VALUE(pv);
+	WT_DECL_RET;
+	WT_PACK_STREAM *ps;
+	WT_SESSION *wt_session;
+	const char *retp, *sep;
+
+	ps = NULL;
+	wt_session = (WT_SESSION *)session;
+
+	WT_ERR(__wt_buf_init(session, buf, 32));
+
+	WT_ERR(wiredtiger_unpack_start(wt_session, format, p, size, &ps));
+
+	for (sep = ""; *format != '\0'; ++format)
```

Comment:
This loop should use `__pack_init`, `__pack_next` rather than attempting to parse the string.  The current version will fail with a format like "3qS", for example.

Reply:
Pushed in 3a4d6ec.